flang/runtime/ISO_Fortran_binding.cpp

```diff
@@ -75,7 +75,7 @@ RT_API_ATTRS int CFI_allocate(CFI_cdesc_t *descriptor,
     dim->sm = byteSize;
     byteSize *= extent;
   }
-  void *p{std::malloc(byteSize)};
+  void *p = byteSize ? std::malloc(byteSize) : std::malloc(1);
```

Comment:
nit: runtime code use braces in all initializations: `void *p{byteSize ? std::malloc(byteSize) : std::malloc(1)};`.

LGTM otherwise.